src/printer.rs

```diff
@@ -125,43 +126,75 @@ fn print_tracks(doc: &mut Document, race: &Race) {
+        return;
+    }
+
+    finished.sort_by(|a, b| a.categories_rank[category].cmp(&b.categories_rank[category]));
+
+    if let Some(limit) = limit {
+        finished.truncate(limit);
+    }
+
+    if limit.is_none() {
+        doc.push(elements::PageBreak::new());
+    }
+
+    let mut table = elements::TableLayout::new(vec![1, 2, 2, 2, 1]);
+
+    print_heading(doc, format!("Category: {}", category.0).as_str());
+    print_table_header(&mut table);
+
+    for racer in finished {
+        table
+            .row()
+            .element(elements::Paragraph::new(format!("{}", racer.start_number)).padded(1))
+            .element(elements::Paragraph::new(&racer.first_name).padded(1))
+            .element(elements::Paragraph::new(&racer.last_name).padded(1))
+            .element(elements::Paragraph::new(format_time_delta_millis(racer.time)).padded(1))
+            .element(
+                elements::Paragraph::new(format!("{}", racer.categories_rank[category])).padded(1),
+            )
+            .push()
+            .expect("Invalid table row");
+    }
+
+    doc.push(table);
+}
+
 fn print_categories(doc: &mut Document, race: &Race) {
     for category in &race.categories {
-        let mut finished: Vec<&Racer> = race
+        let finished: Vec<&Racer> = race
             .racers
             .iter()
             .filter(|r| r.categories.contains(category))
             .filter(|r| r.finish.is_some())
             .collect();
 
-        if finished.is_empty() {
-            continue;
-        }
-
-        finished.sort_by(|a, b| a.categories_rank[category].cmp(&b.categories_rank[category]));
-
-        doc.push(elements::PageBreak::new());
-
-        let mut table = elements::TableLayout::new(vec![1, 2, 2, 2, 1]);
+        print_category_table(doc, category, finished, None);
+    }
+}
 
-        print_heading(doc, format!("Category: {}", category.0).as_str());
-        print_table_header(&mut table);
+fn print_winners(doc: &mut Document, race: &Race) {
+    doc.push(elements::PageBreak::new());
+    print_heading(doc, "Winners");
 
-        for racer in finished {
-            table
-                .row()
-                .element(elements::Paragraph::new(format!("{}", racer.start_number)).padded(1))
-                .element(elements::Paragraph::new(&racer.first_name).padded(1))
-                .element(elements::Paragraph::new(&racer.last_name).padded(1))
-                .element(elements::Paragraph::new(format_time_delta_millis(racer.time)).padded(1))
-                .element(
-                    elements::Paragraph::new(format!("{}", racer.categories_rank[category]))
-                        .padded(1),
-                )
-                .push()
-                .expect("Invalid table row");
-        }
+    for category in &race.categories {
+        let finished: Vec<&Racer> = race
+            .racers
+            .iter()
+            .filter(|r| r.categories.contains(category))
+            .filter(|r| r.finish.is_some())
```

Comment:
its repeated multiple times - maybe introduce a new method (in a race directly)?